src/librustc_typeck/check/mod.rs

```diff
@@ -1939,26 +1939,115 @@ fn check_impl_items_against_trait<'tcx>(
                 missing_items.iter()
                     .map(|trait_item| trait_item.ident.to_string())
                     .collect::<Vec<_>>().join("`, `")));
+
+        // `Span` before impl block closing brace.
```

Comment:
The stuff inside this `if !missing_items...` can be moved into its own function to move error rendering out of the happy path.